R/btw_this.R

```diff
@@ -0,0 +1,64 @@
+btw_this <- function(x, ...) {
+  UseMethod("btw_this")
+}
+
+#' @export
+btw_this.default <- function(x, ...) {
+  # TODO: Replace with {evaluate}
+  out <- capture.output(print(x))
+  if (length(out) && nzchar(out)) return(out)
+
+  capture.output(print(x), type = "message")
+}
+
+#' @export
+btw_this.data.frame <- function(x, ...) {
+  # Implementation for the data.frame method
+}
+
+#' @export
+btw_this.tbl <- function(x, ...) {
+  # Implementation for the tbl method (same as data.frame)
+}
+
+#' @export
+btw_this.Chat <- function(x, ...) {
+  btw_ignore()
+}
+
+#' @export
+btw_this.function <- function(x, ...) {
+  # Implementation for the function method
+}
+
+#' @export
+btw_this.btw_docs_topic <- function(x, ...) {
+  get_help_page(x$package, x$topic)
+}
+
+#' @export
+btw_this.btw_docs_package <- function(x, ...) {
+  get_package_help(x$package)
+}
```

Comment:
Some of these could also use the R's classes, e.g. `?dplyr::glimpse` returns a `help_files_with_topic` class. But we can have our own internal classes too, e.g. `btw_this(as_btw_docs_topic("dplyr", "glimpse"))`

Reply:
If this ends up all living in element-by-element dispatch from `btw(...)`, I wonder if we could make 
* `btw_this.function()` just be `btw_this.btw_docs_topic()` with a fallback if there's no docs topic, (e.g. `btw(mutate)` returns mutate helpfile, but `btw(some_function_i_just_wrote)` just returns the definition) and
* `btw_this.btw_docs_package()` just be one possible path from a character element, e.g. `btw(mtcars, "dplyr")` would `btw_this.btw_docs_package()` (and/or perhaps the intro vignette) for the second element.